torch/utils/cpp_extension.py

```diff
@@ -251,6 +256,14 @@ def __init__(self, *args, **kwargs):
         super(BuildExtension, self).__init__(*args, **kwargs)
         self.no_python_abi_suffix = kwargs.get("no_python_abi_suffix", False)
 
+        self.use_ninja = kwargs.get('use_ninja', False)
+        if self.use_ninja and IS_WINDOWS:
```

Comment:
The reason this line of code in this version of the PR is this way is because in the future I want to support ninja for ahead-of-time cpp extensions on windows, but I want users to be able to switch over to `use_ninja=True` right now without having to special case windows.

Ed's suggestion above in his review is to make `use_ninja=True` by default and have it be 100% a suggestion. I am leaning towards going that route, it would also help with the concern here that "use_ninja" is sometimes a suggestion and sometimes enforced.